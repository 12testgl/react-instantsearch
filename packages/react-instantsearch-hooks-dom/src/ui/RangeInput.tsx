import React from 'react';

import { cx } from './lib/cx';

import type { useRange } from 'react-instantsearch-hooks';

type RangeRenderState = ReturnType<typeof useRange>;

export type RangeInputProps = Omit<
  React.HTMLAttributes<HTMLDivElement>,
  'onSubmit'
> &
  Pick<RangeRenderState, 'range' | 'start'> & {
    classNames?: Partial<RangeInputClassNames>;
    disabled: boolean;
    onSubmit: RangeRenderState['refine'];
    translations: RangeInputTranslations;
  };

export type RangeInputClassNames = {
  /**
   * Class names to apply to the root element
   */
  root: string;
  /**
   * Class names to apply to the root element when there are no refinements possible
   */
  rootNoRefinement: string;
  /**
   * Class names to apply to the form element
   */
  form: string;
  /**
   * Class name to apply to each label element
   */
  label: string;
  /**
   * Class names to apply to each input element
   */
  input: string;
  /**
   * Class names to apply to the minimum input element
   */
  inputMin: string;
  /**
   * Class names to apply to the maximum input element
   */
  inputMax: string;
  /**
   * Class names to apply to the separator element
   */
  separator: string;
  /**
   * Class names to apply to the submit button
   */
  submit: string;
};

export type RangeInputTranslations = {
  /**
   * The label of the separator, between the minimum and maximum inputs
   */
  separator: string;
  /**
   * The label of the submit button
   */
  submit: string;
};

// if the default value is undefined, React considers the component uncontrolled initially, which we don't want 0 or NaN as the default value
const unsetNumberInputValue = '' as unknown as number;

export function RangeInput({
  classNames = {},
  range: { min, max },
  start: [minValue, maxValue],
  disabled,
  onSubmit,
  translations,
  ...props
}: RangeInputProps) {
  const values = {
    min:
      minValue !== -Infinity && minValue !== min
        ? minValue
        : unsetNumberInputValue,
    max:
      maxValue !== Infinity && maxValue !== max
        ? maxValue
        : unsetNumberInputValue,
  };

  const [{ from, to }, setRange] = React.useState({
    from: values.min,
    to: values.max,
  });

  React.useEffect(() => {
    setRange({ from: values.min, to: values.max });
  }, [values.min, values.max]);

  return (
    <div
      {...props}
      className={cx(
        cx('ais-RangeInput', classNames.root),
        disabled &&
          cx('ais-RangeInput-noRefinement', classNames.rootNoRefinement),
        props.className
      )}
    >
      <form
        className={cx('ais-RangeInput-form', classNames.form)}
        onSubmit={(event) => {
          event.preventDefault();
          onSubmit([from, to]);
        }}
      >
        <input
          className={cx(
            cx('ais-RangeInput-input', classNames.input),
            cx('ais-RangeInput-input--min', classNames.inputMin)
          )}
          type="number"
          min={min}
          max={max}
          value={from}
          placeholder={min ? min.toString() : ''}
          disabled={disabled}
          onChange={(event) =>
            setRange({ from: Number(event.currentTarget.value), to })
          }
        />
        <span className={cx('ais-RangeInput-separator', classNames.separator)}>
          {translations.separator}
        </span>
        <input
          className={cx(
            cx('ais-RangeInput-input', classNames.input),
            cx('ais-RangeInput-input--max', classNames.inputMax)
          )}
          type="number"
          min={min}
          max={max}
          value={to}
          placeholder={max ? max.toString() : ''}
          disabled={disabled}
          onChange={(event) =>
            setRange({ from, to: Number(event.currentTarget.value) })
          }
        />
        <button
          className={cx('ais-RangeInput-submit', classNames.submit)}
          type="submit"
        >
          {translations.submit}
        </button>
      </form>
    </div>
  );
}
