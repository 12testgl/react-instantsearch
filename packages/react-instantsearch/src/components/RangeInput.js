import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { isNaN } from 'lodash';
import translatable from '../core/translatable';
import classNames from './classNames.js';

const cx = classNames('RangeInput');

export class RawRangeInput extends Component {
  static propTypes = {
    currentRefinement: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
    canRefine: PropTypes.bool.isRequired,
    translate: PropTypes.func.isRequired,
    refine: PropTypes.func.isRequired,
    min: PropTypes.number,
    max: PropTypes.number,
  };

  static contextTypes = {
    canRefine: PropTypes.func,
  };

  constructor(props) {
    super(props);

    this.state = this.normalizeStateForRendering(props);
  }

  componentWillMount() {
    if (this.context.canRefine) {
      this.context.canRefine(this.props.canRefine);
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.canRefine) {
      this.setState(this.normalizeStateForRendering(nextProps));
    }

    if (this.context.canRefine) {
      this.context.canRefine(nextProps.canRefine);
    }
  }

  onSubmit = e => {
    e.preventDefault();

    this.props.refine({
      min: this.state.from,
      max: this.state.to,
    });
  };

  normalizeStateForRendering(props) {
    const { min: rangeMin, max: rangeMax } = props;
    const { min: valueMin, max: valueMax } = props.currentRefinement;

    return {
      from: valueMin !== undefined && valueMin !== rangeMin ? valueMin : '',
      to: valueMax !== undefined && valueMax !== rangeMax ? valueMax : '',
    };
  }

  normalizeRangeForRendering({ min, max }) {
    const hasMin = min !== undefined;
    const hasMax = max !== undefined;

    return {
      min: hasMin && hasMax ? min : '',
      max: hasMin && hasMax ? max : '',
    };
  }

  render() {
    const { from, to } = this.state;
    const { translate, canRefine } = this.props;
    const { min, max } = this.normalizeRangeForRendering(this.props);

    return (
      <form
        {...cx('root', !canRefine && 'noRefinement')}
        onSubmit={this.onSubmit}
      >
        <fieldset disabled={!canRefine} {...cx('fieldset')}>
          <label {...cx('labelMin')}>
            <input
              {...cx('inputMin')}
              type="number"
              min={min}
              max={max}
              value={from}
              placeholder={min}
              onChange={e => this.setState({ from: e.currentTarget.value })}
            />
          </label>
          <span {...cx('separator')}>{translate('separator')}</span>
          <label {...cx('labelMax')}>
            <input
              {...cx('inputMax')}
              type="number"
              min={min}
              max={max}
              value={to}
              placeholder={max}
              onChange={e => this.setState({ to: e.currentTarget.value })}
            />
          </label>
          <button {...cx('submit')} type="submit">
            {translate('submit')}
          </button>
        </fieldset>
      </form>
    );
  }
}

export default translatable({
  submit: 'ok',
  separator: 'to',
})(RawRangeInput);
