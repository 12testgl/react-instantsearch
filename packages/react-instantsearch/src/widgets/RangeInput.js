import PropTypes from 'prop-types';
import React, { Component } from 'react';
import BaseWidget from './BaseWidget';
import connectRange from '../connectors/connectRange.js';
import RangeInputComponent from '../components/RangeInput.js';
import classNames from '../components/classNames';

const cx = classNames('RangeInput');

/**
 * RangeInput allows a user to select a numeric range using a minimum and maximum input.
 * @name RangeInput
 * @kind widget
 * @requirements The attribute passed to the `attributeName` prop must be holding numerical values.
 * @propType {string} attributeName - the name of the attribute in the record
 * @propType {{min: number, max: number}} [defaultRefinement] - Default state of the widget containing the start and the end of the range.
 * @propType {number} [min] - Minimum value. When this isn't set, the minimum value will be automatically computed by Algolia using the data in the index.
 * @propType {number} [max] - Maximum value. When this isn't set, the maximum value will be automatically computed by Algolia using the data in the index.
 * @propType {number} [precision=2] - Number of digits after decimal point to use.
 * @themeKey ais-RangeInput__root - The root component of the widget
 * @themeKey ais-RangeInput__labelMin - The label for the min input
 * @themeKey ais-RangeInput__inputMin - The min input
 * @themeKey ais-RangeInput__separator - The separator between input
 * @themeKey ais-RangeInput__labelMax - The label for the max input
 * @themeKey ais-RangeInput__inputMax - The max input
 * @themeKey ais-RangeInput__submit - The submit button
 * @themeKey ais-RangeInput__noRefinement - present when there is no refinement
 * @translationKey submit - Label value for the submit button
 * @translationKey separator - Label value for the input separator
 * @example
 * import React from 'react';
 *
 * import { RangeInput, InstantSearch } from 'react-instantsearch/dom';
 *
 * export default function App() {
 *   return (
 *     <InstantSearch
 *       appId="latency"
 *       apiKey="6be0576ff61c053d5f9a3225e2a90f76"
 *       indexName="ikea"
 *     >
 *       <RangeInput attributeName="price"/>
 *     </InstantSearch>
 *   );
 * }
 */

class Widget extends Component {
  componentWillMount() {
    if (this.context.canRefine) {
      this.context.canRefine(this.props.canRefine);
    }
  }

  componentWillReceiveProps(nextProps) {
    if (
      this.context.canRefine &&
      this.props.canRefine !== nextProps.canRefine
    ) {
      this.context.canRefine(nextProps.canRefine);
    }
  }

  render() {
    const { header, footer, canRefine } = this.props;
    return (
      <BaseWidget
        cx={cx}
        header={header}
        footer={footer}
        cantRefine={!canRefine}
      >
        <RangeInputComponent cx={cx} {...this.props} />
      </BaseWidget>
    );
  }
}

Widget.propTypes = {
  canRefine: PropTypes.bool.isRequired,
  header: PropTypes.node,
  footer: PropTypes.node,
};

export default connectRange(Widget);
