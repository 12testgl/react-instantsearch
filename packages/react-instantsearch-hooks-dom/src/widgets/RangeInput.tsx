import React from 'react';
import { useRange } from 'react-instantsearch-hooks';

import { RangeInput as RangeInputUiComponent } from '../ui/RangeInput';

import type { RangeInputProps as RangeInputUiProps } from '../ui/RangeInput';
import type { UseRangeProps } from 'react-instantsearch-hooks';

export type RangeInputProps = Omit<
  RangeInputUiProps,
  'disabled' | 'onSubmit' | 'range' | 'start' | 'translations'
> &
  UseRangeProps;

export function RangeInput({
  attribute,
  min,
  max,
  precision,
  ...props
}: RangeInputProps) {
  const { range, start, canRefine, refine } = useRange(
    { attribute, min, max, precision },
    { $$widgetType: 'ais.rangeInput' }
  );

  return (
    <RangeInputUiComponent
      {...props}
      range={range}
      start={start}
      disabled={!canRefine}
      onSubmit={refine}
      translations={{ separator: 'to', submit: 'Go' }}
    />
  );
}
