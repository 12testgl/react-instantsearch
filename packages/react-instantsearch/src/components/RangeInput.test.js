/* eslint-env jest, jasmine */

import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import serializer from 'enzyme-to-json/serializer';
import RangeInput, { RawRangeInput } from './RangeInput';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(serializer);

describe('RangeInput', () => {
  const shallowRender = (props = {}, context = {}) => {
    const defaultProps = {
      currentRefinement: {
        min: undefined,
        max: undefined,
      },
      canRefine: true,
      refine: () => {},
      min: undefined,
      max: undefined,
    };

    return shallow(<RangeInput {...defaultProps} {...props} />, { context });
  };

  it('render with translations', () => {
    const props = {
      translations: {
        submit: 'SUBMIT',
        separator: 'SEPARATOR',
      },
    };

    const component = shallowRender(props).shallow();

    expect(component).toMatchSnapshot();
  });
});

describe('RawRangeInput', () => {
  const shallowRender = (props = {}, context = {}) => {
    const defaultProps = {
      currentRefinement: {
        min: undefined,
        max: undefined,
      },
      canRefine: true,
      refine: () => {},
      translate: x => x,
      min: undefined,
      max: undefined,
    };

    return shallow(<RawRangeInput {...defaultProps} {...props} />, { context });
  };

  it('render with empty values', () => {
    const props = {};

    const component = shallowRender(props);

    expect(component).toMatchSnapshot();
  });

  it('render with empty values when refinement is equal to min / max', () => {
    const props = {
      currentRefinement: {
        min: 0,
        max: 500,
      },
      min: 0,
      max: 500,
    };

    const component = shallowRender(props);

    expect(component).toMatchSnapshot();
  });

  it('render with refinement', () => {
    const props = {
      currentRefinement: {
        min: 10,
        max: 490,
      },
    };

    const component = shallowRender(props);

    expect(component).toMatchSnapshot();
  });

  it('render with min / max', () => {
    const props = {
      min: 0,
      max: 500,
    };

    const component = shallowRender(props);

    expect(component).toMatchSnapshot();
  });

  it('render with min only', () => {
    const props = {
      min: 0,
    };

    const component = shallowRender(props);

    expect(component).toMatchSnapshot();
  });

  it('render with max only', () => {
    const props = {
      max: 500,
    };

    const component = shallowRender(props);

    expect(component).toMatchSnapshot();
  });

  it("render when can't refine", () => {
    const props = {
      canRefine: false,
    };

    const component = shallowRender(props);

    expect(component).toMatchSnapshot();
  });

  describe('willMount', () => {
    it('expect to call canRefine from context when defined', () => {
      const props = {};
      const context = {
        canRefine: jest.fn(),
      };

      shallowRender(props, context);

      expect(context.canRefine).toHaveBeenCalledTimes(1);
    });

    it('expect to not throw when canRefine is not defined', () => {
      const props = {};
      const context = {};

      expect(() => shallowRender(props, context)).not.toThrow();
    });
  });

  describe('willReceiveProps', () => {
    it('expect to update the state when can refine', () => {
      const props = {};
      const nextProps = {
        canRefine: true,
        currentRefinement: {
          min: 10,
          max: 490,
        },
      };

      const component = shallowRender(props);

      expect(component).toMatchSnapshot();

      component.setProps(nextProps);

      expect(component).toMatchSnapshot();
      expect(component.state()).toEqual({
        from: 10,
        to: 490,
      });
    });

    it('expect to update the state when can refine without refinement', () => {
      const props = {};
      const nextProps = {
        canRefine: true,
        currentRefinement: {},
      };

      const component = shallowRender(props);

      expect(component).toMatchSnapshot();

      component.setProps(nextProps);

      expect(component).toMatchSnapshot();
      expect(component.state()).toEqual({
        from: '',
        to: '',
      });
    });

    it('expect to update the state when can refine and refinement is equal to min / max', () => {
      const props = {};
      const nextProps = {
        canRefine: true,
        min: 10,
        max: 490,
        currentRefinement: {
          min: 10,
          max: 490,
        },
      };

      const component = shallowRender(props);

      expect(component).toMatchSnapshot();

      component.setProps(nextProps);

      expect(component).toMatchSnapshot();
      expect(component.state()).toEqual({
        from: '',
        to: '',
      });
    });

    it("expect to not update the state when can't refine", () => {
      const props = {};
      const nextProps = {
        canRefine: false,
        currentRefinement: {
          min: 10,
          max: 490,
        },
      };

      const component = shallowRender(props);

      expect(component).toMatchSnapshot();

      component.setProps(nextProps);

      expect(component).toMatchSnapshot();
      expect(component.state()).toEqual({
        from: '',
        to: '',
      });
    });

    it('expect to call canRefine from context when defined', () => {
      const props = {};
      const context = {
        canRefine: jest.fn(),
      };

      const nextProps = {
        canRefine: true,
      };

      const component = shallowRender(props, context);

      component.setProps(nextProps);

      expect(context.canRefine).toHaveBeenCalledTimes(2);
    });

    it('expect to not throw when canRefine is not defined', () => {
      const props = {};
      const context = {};
      const nextProps = {};

      const component = shallowRender(props, context);

      expect(() => component.setProps(nextProps)).not.toThrow();
    });
  });

  describe('onChange', () => {
    it('expect to update min onChange', () => {
      const props = {};
      const component = shallowRender(props);

      component.find('.ais-RangeInput__inputMin').simulate('change', {
        currentTarget: {
          value: 10,
        },
      });

      expect(component).toMatchSnapshot();
      expect(component.state()).toEqual({
        from: 10,
        to: '',
      });
    });

    it('expect to update max onChange', () => {
      const props = {};
      const component = shallowRender(props);

      component.find('.ais-RangeInput__inputMax').simulate('change', {
        currentTarget: {
          value: 490,
        },
      });

      expect(component).toMatchSnapshot();
      expect(component.state()).toEqual({
        from: '',
        to: 490,
      });
    });
  });

  describe('onSubmit', () => {
    it('expect to call refine onSubmit with values', () => {
      const props = {
        refine: jest.fn(),
      };

      const event = {
        preventDefault: jest.fn(),
      };

      const component = shallowRender(props);

      component.setState({
        from: 10,
        to: 490,
      });

      component.find('form').simulate('submit', event);

      expect(event.preventDefault).toHaveBeenCalled();
      expect(props.refine).toHaveBeenCalledWith({
        min: 10,
        max: 490,
      });
    });

    it('expect to not call refine with empty string', () => {
      const props = {
        refine: jest.fn(),
      };

      const event = {
        preventDefault: jest.fn(),
      };

      const component = shallowRender(props);

      component.find('form').simulate('submit', event);

      expect(event.preventDefault).toHaveBeenCalled();
      expect(props.refine).not.toHaveBeenCalled();
    });
  });
});
