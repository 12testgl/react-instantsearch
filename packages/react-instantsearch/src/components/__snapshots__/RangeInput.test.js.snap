// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeInput render with translations 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input input--min"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value=""
  />
  <span
    className="separator"
  >
    SEPARATOR
  </span>
  <input
    className="input input--max"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value=""
  />
  <button
    className="submit"
    type="submit"
  >
    SUBMIT
  </button>
</form>
`;

exports[`RawRangeInput onChange expect to update max onChange 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input input--min"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value=""
  />
  <span
    className="separator"
  >
    separator
  </span>
  <input
    className="input input--max"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value={490}
  />
  <button
    className="submit"
    type="submit"
  >
    submit
  </button>
</form>
`;

exports[`RawRangeInput onChange expect to update min onChange 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input input--min"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value={10}
  />
  <span
    className="separator"
  >
    separator
  </span>
  <input
    className="input input--max"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value=""
  />
  <button
    className="submit"
    type="submit"
  >
    submit
  </button>
</form>
`;

exports[`RawRangeInput render when can't refine 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input input--min"
    disabled={true}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value=""
  />
  <span
    className="separator"
  >
    separator
  </span>
  <input
    className="input input--max"
    disabled={true}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value=""
  />
  <button
    className="submit"
    type="submit"
  >
    submit
  </button>
</form>
`;

exports[`RawRangeInput render with empty values 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input input--min"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value=""
  />
  <span
    className="separator"
  >
    separator
  </span>
  <input
    className="input input--max"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value=""
  />
  <button
    className="submit"
    type="submit"
  >
    submit
  </button>
</form>
`;

exports[`RawRangeInput render with empty values when refinement is equal to min / max 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input input--min"
    disabled={false}
    max={500}
    min={0}
    onChange={[Function]}
    placeholder={0}
    step={1}
    type="number"
    value=""
  />
  <span
    className="separator"
  >
    separator
  </span>
  <input
    className="input input--max"
    disabled={false}
    max={500}
    min={0}
    onChange={[Function]}
    placeholder={500}
    step={1}
    type="number"
    value=""
  />
  <button
    className="submit"
    type="submit"
  >
    submit
  </button>
</form>
`;

exports[`RawRangeInput render with max only 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input input--min"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value=""
  />
  <span
    className="separator"
  >
    separator
  </span>
  <input
    className="input input--max"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value=""
  />
  <button
    className="submit"
    type="submit"
  >
    submit
  </button>
</form>
`;

exports[`RawRangeInput render with min / max 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input input--min"
    disabled={false}
    max={500}
    min={0}
    onChange={[Function]}
    placeholder={0}
    step={1}
    type="number"
    value=""
  />
  <span
    className="separator"
  >
    separator
  </span>
  <input
    className="input input--max"
    disabled={false}
    max={500}
    min={0}
    onChange={[Function]}
    placeholder={500}
    step={1}
    type="number"
    value=""
  />
  <button
    className="submit"
    type="submit"
  >
    submit
  </button>
</form>
`;

exports[`RawRangeInput render with min only 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input input--min"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value=""
  />
  <span
    className="separator"
  >
    separator
  </span>
  <input
    className="input input--max"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value=""
  />
  <button
    className="submit"
    type="submit"
  >
    submit
  </button>
</form>
`;

exports[`RawRangeInput render with precision of 1 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input input--min"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={0.1}
    type="number"
    value=""
  />
  <span
    className="separator"
  >
    separator
  </span>
  <input
    className="input input--max"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={0.1}
    type="number"
    value=""
  />
  <button
    className="submit"
    type="submit"
  >
    submit
  </button>
</form>
`;

exports[`RawRangeInput render with precision of 2 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input input--min"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={0.01}
    type="number"
    value=""
  />
  <span
    className="separator"
  >
    separator
  </span>
  <input
    className="input input--max"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={0.01}
    type="number"
    value=""
  />
  <button
    className="submit"
    type="submit"
  >
    submit
  </button>
</form>
`;

exports[`RawRangeInput render with refinement 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input input--min"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value={10}
  />
  <span
    className="separator"
  >
    separator
  </span>
  <input
    className="input input--max"
    disabled={false}
    max=""
    min=""
    onChange={[Function]}
    placeholder=""
    step={1}
    type="number"
    value={490}
  />
  <button
    className="submit"
    type="submit"
  >
    submit
  </button>
</form>
`;
